# backtest-case: BOOT-001+CM-001
# kind: case
# base: 42b840051647eef89761a16dfdff87e82bb253ab
# derived: diff(BASE:transformer_engine/__init__.py, PRE:transformer_engine/__init__.py)
--- a/transformer_engine/__init__.py
+++ b/transformer_engine/__init__.py
@@ -1,3 +1,5 @@
+# This file was modified for portability to AMDGPU
+# Copyright (c) 2025-2026, Advanced Micro Devices, Inc. All rights reserved.
 # Copyright (c) 2022-2026, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
 #
 # See LICENSE for license information.
@@ -8,10 +10,27 @@
 
 import os
 from importlib import metadata
+import os
 import transformer_engine.common
 
+_use_pytorch = True
+_use_jax = True
+
+if os.getenv("NVTE_FRAMEWORK"):
+    _frameworks=os.getenv("NVTE_FRAMEWORK").split(",")
+
+    # Special framework names
+    if "none" in _frameworks:
+        _use_pytorch = False
+        _use_jax = False
+    elif "all" in _frameworks:
+        pass
+    else:
+        _use_pytorch = "pytorch" in _frameworks
+        _use_jax = "jax" in _frameworks
+
 try:
-    from . import pytorch
+    if _use_pytorch: from . import pytorch
 except ImportError:
     pass
 except FileNotFoundError as e:
@@ -38,7 +57,7 @@
             )
 
 try:
-    from . import jax
+    if _use_jax: from . import jax
 except ImportError:
     pass
 except FileNotFoundError as e:
@@ -64,4 +83,9 @@
                 category=RuntimeWarning,
             )
 
-__version__ = str(metadata.version("transformer_engine"))
+try:
+    __version__ = str(metadata.version("transformer_engine"))
+except metadata.PackageNotFoundError:
+    _te_core_installed, _, __version__ = transformer_engine.common.get_te_core_package_info(True)
+    if not _te_core_installed:
+        raise
